### ISSUE TITLE
Check why cats-effect PRs don't include the SCM link

### ISSUE THREAD

**Issue by fthomas** [fthomas/scala-steward#923]
#917 for example does not link to the SCM, but to the cats-effect homepage: https://typelevel.org/cats-effect/. I'm wondering why that is. The POM seems to include the SCM URL:
```xml
    <url>https://typelevel.org/cats-effect/</url>
    <organization>
        <name>Typelevel</name>
        <url>https://typelevel.org/cats-effect/</url>
    </organization>
    <scm>
        <url>https://github.com/typelevel/cats-effect</url>
        <connection>git@github.com:typelevel/cats-effect.git</connection>
    </scm>
```

**Comment by fthomas** [comment]
This is the value of `coursier.core.Info`, that we're getting from Coursier for an update to cats-effect 1.4.0:
```
Info(cats-effect,https://typelevel.org/cats-effect/,List(),List(),None,None)
```
SCM infos are the last field in `Info`, which are `None` in this case.

**Comment by fthomas** [comment]
My next step would be to put the POM into Coursier's test suite to check that it parses it correctly.

**Comment by fthomas** [comment]
Putting https://repo1.maven.org/maven2/org/typelevel/cats-effect_2.12/2.0.0/cats-effect_2.12-2.0.0.pom into `coursier.test.PomParsingTests` yields:
```
Info(cats-effect,https://typelevel.org/cats-effect/,List(...),List(...),None,
Some(Scm(Some(https://github.com/typelevel/cats-effect),Some(git@github.com:typelevel/cats-effect.git),None)))))
```
This looks fine.

**Comment by fthomas** [comment]
I've just checked that https://github.com/coursier/coursier/commit/64e9bc0634e0dcc95b820fd0c5a384c3b8620cbb is in the Coursier version we are using:
```
coursier (master)$ git tag --contains 64e9bc0634e0dcc95b820fd0c5a384c3b8620cbb
v2.0.0-RC3-4
```
We are depending on 2.0.0-RC3-4, so this looks also fine to me. This fix also required code changes, which were done in https://github.com/fthomas/scala-steward/pull/834. So we definitely depend on a Coursier version with the above fix.

**Comment by fthomas** [comment]
[This code](https://github.com/coursier/coursier/blob/0dde4226fad737ea0fa98f799c252db700e5fd50/modules/core/shared/src/main/scala/coursier/maven/PomParser.scala#L234-L241) looks suspicous:
```scala
Info(
  description,
  url,
  Nil, // TODO
  Nil, // TODO
  None,
  None
)
```
The shape of that matches what I've observed in https://github.com/fthomas/scala-steward/issues/923#issuecomment-531379861.

**Comment by fthomas** [comment]
The use of the PomParser is controlled by [useSaxParser](https://github.com/coursier/coursier/blob/0dde4226fad737ea0fa98f799c252db700e5fd50/modules/core/shared/src/main/scala/coursier/maven/MavenRepository.scala):
```
  // only used during benchmarks
  private[coursier] var useSaxParser = true
...
    for {
      str <- fetch(projectArtifact0)
      proj0 <- EitherT(F.point[Either[String, Project]](if (useSaxParser) parseRawPomSax(str) else parseRawPomDom(str)))
    } yield ...
```
So it seems to me that the `scm` field is not present in `Info` because `useSaxParser` is `true` by default.

@exoego If my reasoning above is correct, your changes in Coursier don't have an effect in Scala Steward because `Info.scm` is always set to `None` in `PomParser`.

**Comment by exoego** [comment]
I will look into but stacked due to https://github.com/coursier/coursier/issues/1396

**Comment by exoego** [comment]
Opened a PR https://github.com/coursier/coursier/pull/1403

**Comment by fthomas** [comment]
Fixed by #1036.

**Comment by fthomas** [comment]
Here is a proof that this works now: https://github.com/fthomas/refined-sjs-example/pull/55
